### PR TITLE
ProningAnimation_From_Idle

### LINKED ISSUE #2: Prone from idle (& reverse)
Dad?

### PR DESCRIPTION
**Pull request by ChapeauxdeGoat**
This is way too ez - Proning animation from idle complete!
Closes #2

### CODE REVIEW

File: controllers/PronePlayerState.gd [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+	pass
+
+func update(delta):
+	PLAYER.update_gravity(delta)
+	PLAYER.update_input(SPEED, ACCELERATION, DECELERATION)
+	PLAYER.update_velocity()
+
+	if Input.is_action_just_pressed("prone"):
+		unprone()
+	#elif Input.is_action_pressed("crouch") == false and RELEASED == false:  # When not holding crouch, set crouch_released to true. Then trigger unprone.
```
Comment (lethalgem) [inline controllers/PronePlayerState.gd:32]: Please delete

File: controllers/PronePlayerState.gd [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+		await ANIMATION.animation_finished
+	transition.emit("IdlePlayerState")
+	#if CROUCH_SHAPECAST.is_colliding() == false and Input.is_action_pressed("crouch") == false:
+		#ANIMATION.play("Crouching", -1.0, -PRONE_SPEED * 1.5, true)
+		#if ANIMATION.is_playing():
+			#await ANIMATION.animation_finished
+		#transition.emit("IdlePlayerState")
+	#elif CROUCH_SHAPECAST.is_colliding() == true:
+		#await get_tree().create_timer(0.1).timeout
+		#unprone()
```
Comment (lethalgem) [inline controllers/PronePlayerState.gd:49]:
```suggestion
	#if CROUCH_SHAPECAST.is_colliding() == false and Input.is_action_pressed("crouch") == false:
		#ANIMATION.play("Crouching", -1.0, -PRONE_SPEED * 1.5, true)
		#if ANIMATION.is_playing():
			#await ANIMATION.animation_finished
		#transition.emit("IdlePlayerState")
	#elif CROUCH_SHAPECAST.is_colliding() == true:
		#await get_tree().create_timer(0.1).timeout
		#unprone()
```

**Review by lethalgem — CHANGES_REQUESTED** [review]

File: controllers/PronePlayerState.gd (hunk 1 above)
Comment (lethalgem) [reply controllers/PronePlayerState.gd:32]: the comments not the code

Comment (ChapoTheGoat) [reply controllers/PronePlayerState.gd]: Too ez

File: controllers/PronePlayerState.gd (hunk 2 above)
Comment (ChapoTheGoat) [reply controllers/PronePlayerState.gd]: Way too ez

**Review by lethalgem — APPROVED** [review]

**Review by lethalgem — APPROVED** [review]

**Comment by ChapoTheGoat** [comment]
Let me in!

**Pull request merged**